peerconnection_go_test.go

```diff
@@ -650,3 +650,51 @@ func TestPeerConnection_AnsweringLite(t *testing.T) {
 	assert.NoError(t, offerPC.Close())
 	assert.NoError(t, answerPC.Close())
 }
+
+func TestOnICEGatheringStateChange(t *testing.T) {
+	seenGathering := &atomicBool{}
+	seenComplete := &atomicBool{}
+
+	seenGatheringAndComplete := make(chan interface{})
+	seenClosed := make(chan interface{})
+
+	s := SettingEngine{}
+	s.SetTrickle(true)
+
+	peerConn, err := NewAPI(WithSettingEngine(s)).NewPeerConnection(Configuration{})
+	assert.NoError(t, err)
+
+	peerConn.OnICEGatheringStateChange(func(s ICEGathererState) {
+		switch s {
+		case ICEGathererStateGathering:
+			seenGathering.set(true)
```

Comment:
Fixed!

https://github.com/pion/webrtc/issues/744 has bothered lots of people, but I am not sure of a better way to solve it unfortunately. 

I think eventually it would be fun to explore writing a WebRTC implementation that has a 'Go like API' so we don't have issues like this. Enough work as is though :)